src/main/java/com/springjpa/datajpa/Service/GeoService.java

```diff
@@ -28,49 +33,74 @@
-                } catch (JSONException e) {
-                    e.printStackTrace();
-                }
-                try {
-                    jsonObject = jsonArray.getJSONObject(0);
-                } catch (JSONException e) {
-                    e.printStackTrace();
-                }
-                try {
-                    latitude = jsonObject.getDouble("latitude");
-                } catch (JSONException e) {
-                    e.printStackTrace();
+                ApiResponseDTO response = webClient.get()
+                        .uri(uriBuilder -> uriBuilder
+                                .queryParam("query", "Warszawa")
+                                .build())
+                        .retrieve()
+                        .bodyToMono(ApiResponseDTO.class)
+                        .block();
+                if (response != null) {
+                    Geo newGeo = createGeoFromResponse(response);
+                    System.out.println(newGeo);
+                    geoList.add(newGeo);
                 }
-                try {
-                    longitude = jsonObject.getDouble("longitude");
-                } catch (JSONException e) {
-                    e.printStackTrace();
-                }
-                geoList.add(new Geo(longitude,latitude));
-            });
+            } catch (RuntimeException e) {
+                System.out.println("Error while fetching geo data for station " + name + ": " + e.getMessage());
+            }
+
+//            try {
+//                jsonObject = new JSONObject(geoDTO);
+//            } catch (JSONException e) {
+//                e.printStackTrace();
+//            }
+//            try {
+//                jsonArray = jsonObject.getJSONArray("data");
+//            } catch (JSONException e) {
+//                e.printStackTrace();
+//            }
+//            try {
+//                jsonObject = jsonArray.getJSONObject(0);
+//            } catch (JSONException e) {
+//                e.printStackTrace();
+//            }
+//            try {
+//                latitude = jsonObject.getDouble("latitude");
+//            } catch (JSONException e) {
+//                e.printStackTrace();
+//            }
+//            try {
+//                longitude = jsonObject.getDouble("longitude");
+//            } catch (JSONException e) {
+//                e.printStackTrace();
+//            }
+//            geoList.add(new Geo(longitude, latitude));
+        });
         return geoList;
     }
+
+    private Geo createGeoFromResponse(ApiResponseDTO response) {
+        StationDTO stationDTO = getSingleStationFromData(response);
+        Geo newGeo = new Geo();
+        newGeo.setLatitude(stationDTO.getLatitude());
+        newGeo.setLongitude(stationDTO.getLongitude());
+        return newGeo;
+    }
+
+    private StationDTO getSingleStationFromData(ApiResponseDTO response) {
+        if (!CollectionUtils.isEmpty(response.getData()) && response.getData().size() == 1) {
```

Comment:
Tutaj nie patrzyłem się, co to API zwraca dokładnie. W teorii powinno zawsze jeden element zwracać, tak? Jeśli tak, to można zostawić. Jak zwróci dwa elementry, to może to też jest ok, a nie zapisze ich. Obczaj sobie, jak to dokładnie działa, i jak to 'handlować'